source/MaterialXTest/MaterialXCore/Node.cpp

```diff
@@ -720,8 +729,82 @@ TEST_CASE("Node Definition Creation", "[nodedef]")
             }
         }
         REQUIRE(findDefault);
+
+        doc->removeChild(graph->getName());
+        mx::writeToXmlFile(doc, "definition_from_nodegraph_out.mtlx");
     }
+}
 
-    REQUIRE(doc->validate());
-    mx::writeToXmlFile(doc, "definition_from_nodegraph_out.mtlx");
+TEST_CASE("Functional Graph Interface Change", "[nodedef]")
```

Comment:
Separate into a "clean" test which just tests functional nodegraph interface publishing, removal and modification.